decidim-core/app/models/decidim/action_log.rb

```diff
@@ -55,5 +94,32 @@ def log_presenter_class_for(log_type)
     rescue NameError
       Decidim::Log::BasePresenter
     end
+
+    # Returns a Batchloader for a given class to avoid N+1 queries.
+    #
+    # Since ActionLogs are related to many different resources, loading a collection
+    # of them would trigger a lot of N+1 queries. We're using BatchLoader to
+    # accumulate and group all the resource by their class and only loading them
+    # when it's necessary.
+    def self.lazy_relation(id_method, klass_name, cache)
```

Comment:
This is awesome. It'd be nice if we could abstract this for reuse, but I'm not sure where we have N+1 queries...